src/app.js

```
@@ -1 +1,89 @@
-// write code here
+/* eslint-disable no-console */
+/* eslint-disable strict */
+const fs = require('fs/promises');
+const path = require('path');
+
+async function moveFiles() {
+  const args = process.argv.slice(2);
+
+  if (args.length !== 2) {
+    console.error('Invalid input!');
+
+    return;
+  }
+
+  const [source, destination] = args;
+
+  try {
+    await fs.access(source);
+  } catch (err) {
+    console.error('Source file does not exist!');
+
+    return;
+  }
+
+  // if source & dest is in one folder, we can just rename them
+  if (path.dirname(source) === path.dirname(destination)) {
+    try {
+      await fs.rename(source, destination);
+    } catch (err) {
+      console.error('Error renaming file:', err.message);
+    }
+
+    return;
+  }
+
+  // if destination ends with ('/'), this dir should exists
+  if (destination.endsWith(path.sep)) {
+    try {
+      const stats = await fs.stat(destination);
+
+      if (!stats.isDirectory()) {
+        console.error('Destination is not a directory!');
+
+        return;
+      }
+    } catch (err) {
+      console.error('Destination directory does not exist!');
+
+      return;
+    }
+
+    const fullDestPath = path.join(destination, path.basename(source));
+
+    try {
+      await fs.rename(source, fullDestPath);
+    } catch (err) {
+      console.error('Error moving file:', err.message);
+    }
+
+    return;
+  }
+
+  // if destination exists, and it's a dir
+  try {
+    const stats = await fs.stat(destination);
+
+    if (stats.isDirectory()) {
+      const fullDestPath = path.join(destination, path.basename(source));
+
+      try {
+        await fs.rename(source, fullDestPath);
+      } catch (err) {
+        console.error('Error moving file:', err.message);
+      }
+
+      return;
+    }
+  } catch (err) {
+  }
```

This `try...catch` block has two issues. First, the logic to move a file into a directory (lines 68-74) is a copy of the code in the block above (lines 52-58). This duplication should be avoided. Second, the empty `catch` on line 78 is problematic because it silently ignores all errors from `fs.stat`. This can hide issues like permission errors and make the script behave unpredictably. A better approach is to restructure the logic to determine the final destination path first, and then call `fs.rename` once, which would solve both problems.